### DISCUSSION TITLE
Bug with UUID as primary key

### DISCUSSION THREAD

**Discussion by RChutchev** [spatie/laravel-medialibrary#3756] [Ideas]
Hello, I've tried to install this package but spotted a problem (incompatibility) with UUIDs.
I have a model where IDs are UUIDs, but your migrations use morphs that expect an ID field. 
We're using Filament (v.3) but this problem is not connected to the external package it's on your own.

I'll fix that on my own with uuidMorphs.
Would you be able to put that in future releases?

**Tech details**
Filament 3.2.131
Laravel 11.36.1
PHP 8.3.15
laravel-medialibrary 11.11.0
spatie-laravel-media-library-plugin 3.2.131

The parent model is the Owner (aka property owner) with UUID as the primary ID key (data type is char(36)).
The media table has a UUID on its own, but the external model_id is indicated as "bigint(20) unsigned," which causes a problem.

Screenshot
![Image](https://github.com/user-attachments/assets/fe0defd1-844f-42c5-9f6e-429fd4e01e28)

**Comment by freekmurze** [comment]
Feel free to submit a PR to fix this 👍

**Reply by RChutchev** [reply]
Okay, that's easy to provide a UUID fix by replacing it with the uuidMorphs, but that's also a potential problem with overprovisioning for users who use IDs not UUIDs. IDK maybe the only way is to create an installation wizard where users select the model primary key data type where the default is ordinary IDs. As a maintainer, can you give an advice on which way is better to use for this package?
